### PR TITLE
[Keymap] Corodiaks Kyria

### PR DESCRIPTION
**Pull request by corodiak**
## Description

- Switch between QWERTY and COLEMAK
- Toggle between Windows and macOS without having duplicate layers with macros for each OS

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+            break;
+        case Colemak:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_COLEMAK);
+            }
+            break;
+        case Undo:
+            if (record->event.pressed) {
+                if (user_config.osIsWindows == 1) {
+                    SEND_STRING(SS_LCTL("z"));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(C(KC_Z));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                } else if (user_config.osIsWindows == 0) {
+                    SEND_STRING(SS_LGUI("z"));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(G(KC_Z));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+                }
+            }
+            break;
+        case Cut:
+            if (record->event.pressed) {
+                if (user_config.osIsWindows == 1) {
+                    SEND_STRING(SS_LCTL("x"));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(C(KC_X));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+                } else if (user_config.osIsWindows == 0) {
+                    SEND_STRING(SS_LGUI("x"));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(G(KC_X));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+                }
+            }
+            break;
+        case Copy:
+            if (record->event.pressed) {
+                if (user_config.osIsWindows == 1) {
+                    SEND_STRING(SS_LCTL("c"));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(C(KC_C));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+                } else if (user_config.osIsWindows == 0) {
+                    SEND_STRING(SS_LGUI("c"));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(G(KC_C));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+                }
+            }
+            break;
+        case Paste:
+            if (record->event.pressed) {
+                if (user_config.osIsWindows == 1) {
+                    SEND_STRING(SS_LCTL("v"));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(C(KC_V));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+                } else if (user_config.osIsWindows == 0) {
+                    SEND_STRING(SS_LGUI("v"));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(G(KC_V));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+                }
+            }
+            break;
+        case PrvWord:
+            if (record->event.pressed) {
+                if (user_config.osIsWindows == 1) {
+                    SEND_STRING(SS_LCTL(SS_TAP(X_LEFT())));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(C(KC_LEFT));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+                } else if (user_config.osIsWindows == 0) {
+                    SEND_STRING(SS_LALT(SS_TAP(X_LEFT())));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(A(KC_LEFT));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+                }
+            }
+            break;
+        case NxtWord:
+            if (record->event.pressed) {
+                if (user_config.osIsWindows == 1) {
+                    SEND_STRING(SS_LCTL(SS_TAP(X_RIGHT())));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(C(KC_RGHT));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 12]
```diff
@@ -0,0 +1,321 @@
...
+                } else if (user_config.osIsWindows == 0) {
+                    SEND_STRING(SS_LALT(SS_TAP(X_RIGHT())));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                    tap_code16(A(KC_RGHT));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 13]
```diff
@@ -0,0 +1,321 @@
...
+            user_config.osIsWindows = true;
+            eeconfig_update_user(user_config.raw);
+        }
+
+        // Screenshot
+        SEQ_ONE_KEY(KC_S) {
+            if (user_config.osIsWindows == 1) {
+                SEND_STRING(SS_LGUI("\nS"));
+            } else if (user_config.osIsWindows == 0) {
+                SEND_STRING(SS_LGUI(SS_LSFT("4")));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                tap_code16(S(G(KC_4)));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 14]
```diff
@@ -0,0 +1,321 @@
...
+            }
+        }
+
+        // Video
+        SEQ_ONE_KEY(KC_V) {
+            if (user_config.osIsWindows == 0) {
+                SEND_STRING(SS_LGUI(SS_LSFT("5")));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                tap_code16(S(G(KC_5)));
```

File: keyboards/kyria/keymaps/corodiak/keymap.c [hunk 15]
```diff
@@ -0,0 +1,321 @@
...
+            }
+        }
+
+        // Sleep
+        SEQ_ONE_KEY(KC_P) {
+            if (user_config.osIsWindows == 1) {
+                SEND_STRING(SS_LGUI("x") "u" "h");
+            } else if (user_config.osIsWindows == 0) {
+                SEND_STRING(SS_LGUI(SS_LALT(SS_TAP(X_PWR()))));
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/corodiak/keymap.c]:
```suggestion
                tap_code16(A(G(KC_PWR)));
```

**Comment by corodiak** [comment]
Thanks for the suggestion, I didn't know about this way

**Comment by noroadsleft** [comment]
Thanks!

Also:

> * Toggle between Windows and macOS without having duplicate layers with macros for each OS

Don't mind me if I steal that functionality later. :smiley:

**Comment by corodiak** [comment]
Please do, it's really convenient. 
Especially if you also want to switch between multiple layouts.
I could save 3 layers with this solution

**Pull request merged**